code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -244,7 +244,8 @@
 /datum/effect_system/foam_spread/start()
 	var/obj/effect/particle_effect/foam/F = new effect_type(location)
 	var/foamcolor = mix_color_from_reagents(chemholder.reagents.reagent_list)
-	chemholder.reagents.copy_to(F, chemholder.reagents.total_volume/amount)
+	var/effective_amount = chemholder.reagents.total_volume / max(amount, 3)
```

Comment:
Can you explain what you're doing here and why this fixes it?

Reply:
`amount` is radius of the foam,
for example, 10u foam has radius of 3,
5u foam has radius of 2
1u foam has radius of 1

here I am capping amount of reagent foam recieves by limiting how low it can go,
any radius of foam less than 3 makes foam recieve same amount of reagents as foam of radius 3
maximum multiplication of reagents is about 166% (3 times as low as before, it was about 500% with 1u foam)

Reply:
Can you add this as a comment so it's not lost? (Even though I'd rather we just kill foam duplication for the same reason oranges has stated before)

Reply:
I'll add comment, but this is just quick fix for insane duping while Oranges works on proper fix

Reply:
yeah i dont think hes doing anything